dartagnan/src/main/java/com/dat3m/dartagnan/witness/WitnessBuilder.java

```diff
@@ -124,6 +126,26 @@ private void buildGraph() {
 					edge.addAttribute(CREATETHREAD.toString(), valueOf(threads));
 					threads++;
 				}
+				
+				if(e instanceof Load) {
+					Load l = (Load)e;
+					edge.addAttribute(EVENTID.toString(), valueOf(e.getUId()));
+					// JavaSMT toString() method returns negative numbers as (- X)
+					// We convert this to -X
+					String value = valueOf(model.eval(l.getResultRegisterExpr()));
+					value = value.startsWith("(") && value.endsWith(")") ? value.substring(1, value.length()-1).replace(" ", "") : value;
+					edge.addAttribute(LOADEDVALUE.toString(), value);
+				}
+
+				if(e instanceof Store) {
+					Store s = (Store)e;
+					edge.addAttribute(EVENTID.toString(), valueOf(e.getUId()));
+					// JavaSMT toString() method returns negative numbers as (- X)
+					// We convert this to -X
+					String value = valueOf(model.eval(s.getMemValueExpr()));
```

Comment:
Similarly to the case if ```Load```, you can try to user ```s.getMemValue().getIntValue(...)```. In case that doesn't work, you probably should fix that method to work then :).